src/instructlab/model/train.py

```diff
@@ -521,20 +615,333 @@ def get_files(directory: str, pattern: str) -> list[str]:
+                )
+
+        else:
+            if not os.path.isfile(data_path):
+                ctx.fail(
+                    f"Data path must be to a valid .jsonl file. Value given: {data_path}"
+                )
+
             run_training(train_args=train_args, torch_args=torch_args)
-        # unsure what types of exceptions training library returns, this will catch all for now
-        except Exception as exc:
+
+
+def _prepare_phased_base_dir(phased_base_dir: pathlib.Path) -> None:
+    """Adds phase1 and phase2 directories and subdirectories in phased_base_dir.
+
+    Args:
+        phased_base_dir: directory wrapping phase1 and phase2 cached data.
+    """
+
+    logger.debug(f"Phased Training -- Preparing phased base dir: {phased_base_dir}")
+
+    phase1_dir_path = phased_base_dir / "phase1"
+    phase2_dir_path = phased_base_dir / "phase2"
+
+    for p in [phase1_dir_path, phase2_dir_path]:
+        utils.clear_directory(p)
+        _setup_phase_dirs(p)
+
+
+def _setup_phase_dirs(path: pathlib.Path) -> None:
+    """Creates {path}/checkpoints and {path}/eval_cache directories."""
+    # TODO: these sub-directory names are hard-coded here but they
+    # could be parameterized in config.
+    logger.debug(f"Phased Training -- Created phase directories for {path}")
+    ckpt_path = path / "checkpoints"
+    eval_cache_path = path / "eval_cache"
+
+    os.makedirs(ckpt_path)
+    os.makedirs(eval_cache_path)
+
+
+def _training_phase(
+    train_args: TrainingArgs,
+    torch_args: TorchrunArgs,
+    data_path: pathlib.Path,
+    model_override: pathlib.Path | None = None,
+    ckpt_dir: pathlib.Path | None = None,
+) -> None:
+    """A single step of end-to-end training."""
+
+    # Third Party
+    from instructlab.training import run_training  # pylint: disable=no-name-in-module
+
+    # override whatever got injected at a higher level
+    # NOTE: have to cast pathlib.Path to str because Pydantic models require this.
+    logger.debug(
+        f"Phased Training -- training phase -- Overriding data_path: {train_args.data_path} with {data_path}"
+    )
+    train_args.data_path = str(data_path)
+
+    if ckpt_dir:
+        train_args.ckpt_output_dir = str(ckpt_dir)
+
+    if model_override:
+        logger.debug(
+            f"Phased Training -- training phase -- Overriding model_path: {train_args.model_path} with {model_override}"
+        )
+        train_args.model_path = str(model_override)
+
+    logger.debug(
+        f"Phased Training -- training phase -- Current training args for phase: {pprint.pformat(train_args)}"
+    )
+
+    run_training(train_args=train_args, torch_args=torch_args)
+
+
+def _mmlu(ckpt_path: pathlib.Path) -> float:
+    # Third Party
+    from instructlab.eval.mmlu import MMLU_TASKS, MMLUEvaluator
+    import torch
+
+    evaluator = MMLUEvaluator(ckpt_path, tasks=MMLU_TASKS, few_shots=5)
+
+    mmlu_results: tuple[float, dict] = evaluator.run()
```

Comment:
nit: if you don't care about both values, you `ckpt_score, _ = evaluator.run()`

Reply:
fixed. I didn't love that run() doesn't type-hint the expanded tuple